src/main/java/frc/trigon/robot/motorsimulation/MotorSimulation.java

```diff
@@ -0,0 +1,145 @@
+package frc.trigon.robot.motorsimulation;
+
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.VoltageOut;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Notifier;
+import edu.wpi.first.wpilibj.Timer;
+import frc.trigon.robot.constants.RobotConstants;
+import frc.trigon.robot.utilities.Conversions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A class that simulates a motor and its PID and feedforward.
+ */
+public abstract class MotorSimulation {
+    private static final List<MotorSimulation> REGISTERED_SIMULATIONS = new ArrayList<>();
+
+    static {
+        new Notifier(MotorSimulation::updateRegisteredSimulations).startPeriodic(RobotConstants.PERIODIC_TIME_SECONDS);
+    }
+
+    private PositionVoltage positionVoltageRequest = null;
+    private MotionMagicVoltage motionMagicRequest = null;
+    private PIDController pidController;
+    private ProfiledPIDController profiledPIDController;
+    private MotorSimulationConfiguration config = new MotorSimulationConfiguration();
+    private double voltage = 0;
+
+    protected MotorSimulation() {
+        REGISTERED_SIMULATIONS.add(this);
+    }
+
+    private static void updateRegisteredSimulations() {
+        for (MotorSimulation motorSimulation : REGISTERED_SIMULATIONS) {
+            motorSimulation.updateMotor();
+            if (motorSimulation.motionMagicRequest != null)
+                motorSimulation.setControl(motorSimulation.motionMagicRequest);
+            else if (motorSimulation.positionVoltageRequest != null)
+                motorSimulation.setControl(motorSimulation.positionVoltageRequest);
```

Comment:
Extract this into a non-static function called updateSimulation() in the class, that'll be called here.